lib/os/mem_blocks.c

```diff
@@ -454,3 +455,29 @@ int sys_mem_blocks_runtime_stats_reset_max(sys_mem_blocks_t *mem_block)
 	return 0;
 }
 #endif
+
+#ifdef CONFIG_OBJ_CORE_SYS_MEM_BLOCKS
+static struct k_obj_type obj_type_sys_mem_blocks;
+
+static int init_sys_mem_blocks_module(void)
+{
+	/* Initialize the sys_mem_blocks object type */
+
+	z_obj_type_init(&obj_type_sys_mem_blocks, K_OBJ_TYPE_MEM_BLOCK_ID,
+			offsetof(struct sys_mem_blocks, obj_core));
+
+	/* Initialize statically defined sys_mem_blocks */
+
+	STRUCT_SECTION_FOREACH_ALTERNATE(sys_mem_blocks_ptr,
+					 sys_mem_blocks *, block_pp) {
+		k_obj_core_init(K_OBJ_CORE(*block_pp),
+				&obj_type_sys_mem_blocks);
+		k_obj_core_link(K_OBJ_CORE(*block_pp));
```

Comment:
All these pairs `k_obj_core_init` and `k_obj_core_link` makes me wonder why not a `k_obj_core_init_and_link` one...

Reply:
Seems reasonable.